tools/flight_recorder/components/utils.py

```diff
@@ -216,16 +230,46 @@ def check_version(versions: Dict[str, Any]) -> None:
         # assert minor >= 0, f"Rank {rank} unsupported version {version}"
 
 
-# TODO: We need to revisit this function to see if we still need it.
-def check_trace_from_beginning(entries: Dict[str, Any]) -> bool:
+def sort_trace_from_beginning(
+    entries: Dict[int, List[Dict[str, Any]]]
+) -> Dict[int, List[Dict[str, Any]]]:
+    """
+    Sorts the trace entries by record ID for entries.
+    This function takes a dictionary of rank names to lists of trace entries as input.
+    Each trace entry is a dictionary containing information about a collective operation,
+    including its unique identifier (`record_id` is monotonically increasing as we write into the ring buffer).
+    The function first sorts the entries in each rank by their `record_id` values.
+    Then, it finds the largest starting point across all ranks by taking the maximum
+    `record_id` value of the first entry in each rank. Finally, it filters out any
+    entries with `record_id` values less than the maximum starting point.
+    The function returns the updated dictionary of sorted and filtered trace entries.
+
+    Args:
+        entries (Dict[str, List[Dict[str, Any]]]): A dictionary of rank names to lists of trace entries.
+
+    Returns:
+        entries (Dict[str, List[Dict[str, Any]]]): Entries sorted by record ID and filtered by the maximum starting point.
+    """
+
+    maximum_starting_record_id = 0
     for rank in entries:
+        # Since this is a ring buffer, we want to first sort the entries by `record_id` to
+        # find the largest starting point. For example, if the buffer has the following entries:
+        # Rank 0: [0, 1, 2, 3, 4, 5, 6]
+        # Rank 1: [7, 1, 2, 3, 4, 5, 6]
+        # Rank 2: [7, 8, 2, 3, 4, 5, 6]
```

Comment:
One thing to note is the expectation is entry `0` is the lowest record id and entry `N` is the largest record id. FR entries are supposed to be sorted when dumped.
So, we should not see 7, 8, 1, 2, 3 .

Reply:
Oh right, I missed that part, inside `dump_entries` we already have sorting already, then we don't need to re-sort it again.